shared/src/main/kotlin/com/egm/stellio/shared/util/ApiUtils.kt

```diff
@@ -86,3 +88,37 @@ fun parseAndExpandAttrsParameter(attrsParam: String?, contextLink: String): Set<
             JsonLdUtils.expandJsonLdKey(it.trim(), contextLink)!!
         }
         .toSet()
+
+fun getApplicableMediaType(httpHeaders: HttpHeaders): MediaType =
+    httpHeaders.accept.getApplicable()
+
+/**
+ * Return the applicable media type among JSON_LD_MEDIA_TYPE and MediaType.APPLICATION_JSON.
+ *
+ * @throws NotAcceptableStatusException if none of the above media types are applicable
+ */
+fun List<MediaType>.getApplicable(): MediaType {
+    if (this.isEmpty())
+        return MediaType.APPLICATION_JSON
+    MediaType.sortByQualityValue(this)
+    val mediaType = this.find {
+        it.includes(MediaType.APPLICATION_JSON) || it.includes(JSON_LD_MEDIA_TYPE)
+    } ?: throw NotAcceptableStatusException(listOf(MediaType.APPLICATION_JSON, JSON_LD_MEDIA_TYPE))
```

Comment:
Isn't already done with the @Produces Annotation in the Handlers ?

Reply:
yes we should not fall into the `throw` case ... but I was worried about not handling the case where no `mediaType` is found.